include/deal.II/fe/fe_enriched.h

```diff
@@ -644,6 +648,160 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
+   * function associated with <code> FE_Q<2>(1) </code> should be same since
+   * they share boundary and Helper class ensures this.
+   *
+   * <table>
+   * <tr>
+   * <th>Active FE index</th>
+   * <th>FiniteElement name</code> </th>
+   * </tr>
+   * <tr>
+   * <td>0</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Nothing<2>()-FE_Nothing<2>()]</code></td>
+   * </tr>
+   * <tr>
+   * <td>1</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Q<2>(1)-FE_Nothing<2>()]</code></td>
+   * </tr>
+   * <tr>
+   * <td>2</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Q<2>(1)-FE_Q<2>(1)]</code></td>
+   * </tr>
+   * <tr>
+   * <td>3</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Nothing<2>()-FE_Q<2>(1)]</code></td>
+   * </tr>
+   * </table>
+   *
+   * The internal data members used by this class need to be available when the
+   * problem is solved. This can be ensured by declaring the object static,
+   * which then remains in the memory until the end of program. An alternative
+   * would be to use it as a data member of the containing class. Since vector
+   * of predicates and enrichment functions may not be available while
+   * constructing the Helper<dim>, a shared_ptr<Helper<dim>> can be used and
+   * constructed in place i.e when the predicates and enrichment functions are
+   * available.
+   *
+   * Example usage:
+   * @code
+   * FE_Q<dim> fe_base(2);
+   * FE_Q<dim> fe_enriched(1);
+   * std::vector< predicate_function<dim> > vec_predicates;
+   * std::vector< std::shared_ptr<Function<dim>> > vec_enrichments;
+   * hp::FECollection<dim> fe_collection;
+   *
+   * Triangulation<dim>  triangulation;
+   * hp::DoFHandler<dim> dof_handler(triangulation);
+   *
+   * static ColorEnriched::Helper<dim> FE_helper(fe_base,
+   *                                             fe_enriched,
+   *                                             vec_predicates,
+   *                                             vec_enrichments);
+   * FE_helper.set(dof_handler);
+   * fe_collection = FE_helper.get_fe_collection();
+   * @endcode
+   */
+  template<int dim>
+  struct Helper
+  {
+    /**
+     * A simple constructor for Helper class.
+     *
+     * @param fe_base base FiniteElement
+     * @param fe_enriched enriched FiniteElement
+     * @param vec_predicates vector of predicates defining the subdomains.
+     * <code>@p vec_predicates[i]</code> returns true for a cell if it
+     * belongs to subdomain with index i.
+     * @param vec_enrichments vector of enrichment functions
+     */
+    Helper(const FE_Q<dim> &fe_base,
+           const FE_Q<dim> &fe_enriched,
+           const std::vector<predicate_function<dim>> &vec_predicates,
+           const std::vector<std::shared_ptr<Function<dim>>> &vec_enrichments);
+
+    /**
+     * Prepares @p dof_handler by modifying material id of cells and
+     * constructs data members cellwise_color_predicate_map and fe_sets which
```

Comment:
do not expose internal implementation. in `public: ` we shall just explain what this does and make sure users understand when this is to be called. How is it implemented internally and which private members are being initialized here is not important.